### PR TITLE
fix: add logging to ES TraceWriter for silent write failures

### LINKED ISSUE #8058: [Bug]: Archive trace returns 200 but does not write to Elasticsearch
### What happened?

Clicking "Archive" on a trace in the Jaeger UI shows "This trace has been archived", but nothing is written to Elasticsearch. No archive indices are created and no errors are logged.

Calling the API directly confirms the issue — it returns HTTP 200 with an empty response that looks like a search result, not a write acknowledgment:

```
POST /api/archive/{traceID}
→ 200 {"data":[],"total":0,"limit":0,"offset":0,"errors":[]}
```

### Steps to reproduce

1. Deploy Jaeger v2.14.1 with Elasticsearch and `traces_archive` configured (see config below)
2. Open a trace in the Jaeger UI
3. Click the "Archive" button
4. UI shows "This trace has been archived"
5. Check Elasticsearch — no `archive*` indices exist

### Expected behavior

The trace should be written to Elasticsearch under the archive index prefix, and `archive-jaeger-span-*` indices should appear.

### Relevant log output

```shell
# No archive-related log output at all, even after clicking Archive.
# Startup logs show both stores initializing successfully:
Initializing storage 'archive_store_elasticsearch'
Elasticsearch detected  version=7
Initializing storage 'primary_store_elasticsearch'
Elasticsearch detected  version=7
```

### Additional context

Primary storage works fine — `jaeger-span-*` and `jaeger-service-*` indices are created and populated. Only the archive write is silently failing.

Possibly related to #6225 (storage role assignment).

## Jaeger setup

**Jaeger backend version:** v2.14.1

**Storage backend:** AWS OpenSearch 7.x (VPC endpoint)

**Deployment configs:**

```shell
extensions:
  jaeger_query:
    storage:
      traces: primary_store_elasticsearch
      traces_archive: archive_store_elasticsearch
  jaeger_storage:
    backends:
      primary_store_elasticsearch:
        elasticsearch:
          server_urls: ["https://my-opensearch.eu-west-1.es.amazonaws.com:443"]
      archive_store_elasticsearch:
        elasticsearch:
          server_urls: ["https://my-opensearch.eu-west-1.es.amazonaws.com:443"]
          indices:
            index_prefix: "archive"
```

### PR DESCRIPTION
**Pull request by majiayu000**
## Summary
- Add logger to `TraceWriter` struct using existing `Logger` from `SpanWriterParams`
- Log a warning when `ToDBModel` returns 0 spans from non-empty trace data, making silent archive write failures diagnosable
- Log a debug message on successful span writes to ES
- Add test for empty traces case (`WriteTraces` with zero `ResourceSpans`)

Fixes #8058

## Test plan
- [x] New `TestTraceWriter_WriteTraces_EmptyTraces` test verifies behavior with empty traces
- [x] Existing `TestTraceWriter_WriteTraces` updated to include logger
- [x] `go test ./internal/storage/v2/elasticsearch/tracestore/...` — all pass
- [x] `make fmt` — clean
- [x] `make lint` — 0 issues
- [x] Full test suite (`go test -race ./...`) — all pass

### CODE REVIEW

File: internal/storage/v2/elasticsearch/tracestore/writer_test.go [hunk 1]
```diff
@@ -26,16 +26,25 @@ func TestTraceWriter_WriteTraces(t *testing.T) {
...
 }
 
+func TestTraceWriter_WriteTraces_EmptyTraces(t *testing.T) {
+	coreWriter := &mocks.CoreSpanWriter{}
+	writer := TraceWriter{spanWriter: coreWriter, logger: zap.NewNop()}
+	td := ptrace.NewTraces()
+	err := writer.WriteTraces(context.Background(), td)
+	require.NoError(t, err)
+	coreWriter.AssertNotCalled(t, "WriteSpan")
+}
```
[the remark this answers is not in the document]
Comment (majiayu000) [reply internal/storage/v2/elasticsearch/tracestore/writer_test.go:75]: Done. Tests now use `zaptest/observer` to assert log level and message for all three paths: (1) empty traces → debug, no warn, (2) non-empty ResourceSpans with zero converted spans → warn with counts, (3) successful write → debug. Fixed in latest commit.

File: internal/storage/v2/elasticsearch/tracestore/writer.go [hunk 2]
```diff
@@ -7,29 +7,39 @@ import (
...
+		logger:     p.Logger,
 	}
 }
 
 // WriteTraces convert the traces to ES Span model and write into the database
 func (t *TraceWriter) WriteTraces(_ context.Context, td ptrace.Traces) error {
 	dbSpans := ToDBModel(td)
+	if len(dbSpans) == 0 {
+		t.logger.Warn("no spans converted from trace data",
+			zap.Int("resource_spans", td.ResourceSpans().Len()),
```
[the remark this answers is not in the document]
Comment (majiayu000) [reply internal/storage/v2/elasticsearch/tracestore/writer.go]: Done. Restructured `WriteTraces` to return early (with debug log) when `ResourceSpans` is empty — avoids calling `ToDBModel` unnecessarily. The warning now only fires for non-empty trace payloads that produce 0 DB spans, and includes `scope_spans` and `spans` counts for better debugging.

File: internal/storage/v2/elasticsearch/tracestore/writer.go [hunk 3]
```diff
@@ -7,29 +7,39 @@ import (
...
 type TraceWriter struct {
 	spanWriter spanstore.CoreSpanWriter
+	logger     *zap.Logger
 }
 
 // NewTraceWriter returns the TraceWriter for use
 func NewTraceWriter(p spanstore.SpanWriterParams) *TraceWriter {
 	return &TraceWriter{
 		spanWriter: spanstore.NewSpanWriter(p),
+		logger:     p.Logger,
```
[the remark this answers is not in the document]
Comment (majiayu000) [reply internal/storage/v2/elasticsearch/tracestore/writer.go]: Done. `NewTraceWriter` now defaults to `zap.NewNop()` when `p.Logger` is nil, preventing nil-pointer panics.

File: internal/storage/v2/elasticsearch/tracestore/writer_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (majiayu000) [reply internal/storage/v2/elasticsearch/tracestore/writer_test.go:75]: Done — tests now use `zaptest/observer` to assert log level and message for all three cases: (1) empty traces → debug log "skipping write of empty trace data", no warn; (2) non-empty ResourceSpans with zero converted spans → warn "no spans converted from trace data" with resource_spans/scope_spans/spans counts; (3) successful write → debug log "wrote spans to ES". See `TestTraceWriter_WriteTraces_EmptyTraces` and `TestTraceWriter_WriteTraces_NonEmptyResourceSpansZeroSpans`.

File: internal/storage/v2/elasticsearch/tracestore/writer.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (majiayu000) [reply internal/storage/v2/elasticsearch/tracestore/writer.go]: Done — `WriteTraces` now returns early with a debug log (no warn) when `td.ResourceSpans().Len() == 0`. The warn with detailed counts (resource_spans, scope_spans, spans) only fires when non-empty trace payloads produce 0 DB spans after conversion.

File: internal/storage/v2/elasticsearch/tracestore/writer.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (majiayu000) [reply internal/storage/v2/elasticsearch/tracestore/writer.go]: Done — `NewTraceWriter` now defaults to `zap.NewNop()` when `p.Logger` is nil, preventing nil-pointer panics.

File: internal/storage/v2/elasticsearch/tracestore/writer.go [hunk 4]
```diff
@@ -7,29 +7,60 @@ import (
...
 // NewTraceWriter returns the TraceWriter for use
 func NewTraceWriter(p spanstore.SpanWriterParams) *TraceWriter {
+	logger := p.Logger
+	if logger == nil {
+		logger = zap.NewNop()
+	}
 	return &TraceWriter{
 		spanWriter: spanstore.NewSpanWriter(p),
+		logger:     logger,
 	}
```
[the remark this answers is not in the document]
Comment (majiayu000) [reply internal/storage/v2/elasticsearch/tracestore/writer.go:31]: Done. `NewTraceWriter` now updates `p.Logger` before passing to `spanstore.NewSpanWriter(p)`, ensuring the inner SpanWriter also gets a non-nil logger. Added `Test_NewTraceWriter_NilLogger` to verify.

File: cmd/jaeger/internal/extension/jaegerquery/querysvc/service.go [hunk 5]
```diff
@@ -165,6 +174,9 @@ func (qs QueryService) ArchiveTrace(ctx context.Context, query tracestore.GetTra
 	if qs.options.ArchiveTraceWriter == nil {
 		return errNoArchiveSpanStorage
 	}
+	qs.logger.Debug("archive trace request received",
+		zap.String("trace_id", query.TraceID.String()),
+	)
```
[the remark this answers is not in the document]
Comment (majiayu000) [reply cmd/jaeger/internal/extension/jaegerquery/querysvc/service.go:179]: `NewQueryService` already defaults `qs.logger` to `zap.NewNop()` when `options.Logger` is nil (lines 81-84). Since `QueryService` fields are unexported, it can only be constructed via `NewQueryService`, so `qs.logger` is guaranteed non-nil in `ArchiveTrace`. No additional guard needed.

File: internal/storage/v2/elasticsearch/tracestore/writer.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (majiayu000) [reply internal/storage/v2/elasticsearch/tracestore/writer.go:31]: Done. `NewTraceWriter` now updates `p.Logger` before passing to `spanstore.NewSpanWriter(p)`, ensuring the inner SpanWriter also gets a non-nil logger. Added `Test_NewTraceWriter_NilLogger` to verify.

File: cmd/jaeger/internal/extension/jaegerquery/querysvc/service.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (majiayu000) [reply cmd/jaeger/internal/extension/jaegerquery/querysvc/service.go:179]: `NewQueryService` already defaults `qs.logger` to `zap.NewNop()` when `options.Logger` is nil (lines 81-84). Since `QueryService` fields are unexported, it can only be constructed via `NewQueryService`, so `qs.logger` is guaranteed non-nil in `ArchiveTrace`. No additional guard needed.

File: internal/storage/v2/elasticsearch/tracestore/writer.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (majiayu000) [reply internal/storage/v2/elasticsearch/tracestore/writer.go:31]: Done. `NewTraceWriter` now updates `p.Logger` before passing to `spanstore.NewSpanWriter(p)`, ensuring the inner SpanWriter also gets a non-nil logger. Added `Test_NewTraceWriter_NilLogger` to verify.

File: cmd/jaeger/internal/extension/jaegerquery/querysvc/service.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (majiayu000) [reply cmd/jaeger/internal/extension/jaegerquery/querysvc/service.go:179]: `NewQueryService` already defaults `qs.logger` to `zap.NewNop()` when `options.Logger` is nil (lines 81-84). Since `QueryService` fields are unexported, it can only be constructed via `NewQueryService`, so `qs.logger` is guaranteed non-nil in `ArchiveTrace`. No additional guard needed.
